usaspending_api/etl/es_etl_helpers.py

```diff
@@ -452,43 +446,47 @@ def csv_chunk_gen(filename, chunksize, job_id, load_type):
         # and this is needed for performance
         # ES helper will pop any "meta" fields like "routing" from provided data dict and use them in the action
         file_df["routing"] = file_df[settings.ES_ROUTING_FIELD]
+
+        # Explicitly setting the ES _id field to match the postgres PK value allows
+        # bulk index operations to be upserts without creating duplicate documents
+        file_df["_id"] = file_df[f"{'award' if load_type == 'awards' else 'transaction'}_id"]
         yield file_df.to_dict(orient="records")
 
 
 def es_data_loader(client, fetch_jobs, done_jobs, config):
     if config["create_new_index"]:
         # ensure template for index is present and the latest version
-        call_command("es_configure", "--template-only", "--load_type={}".format(config["load_type"]))
+        call_command("es_configure", "--template-only", f"--load-type={config['load_type']}")
     while True:
         if not done_jobs.empty():
             job = done_jobs.get_nowait()
             if job.name is None:
                 break
 
-            printf({"msg": "Starting new job", "job": job.name, "f": "ES Ingest"})
+            printf({"msg": "Starting new job", "job": job.name, "f": "ES Index"})
             post_to_elasticsearch(client, job, config)
             if os.path.exists(job.csv):
                 os.remove(job.csv)
         else:
-            printf({"msg": "No Job. Sleeping 45s", "f": "ES Ingest"})
-            sleep(45)
+            printf({"msg": f"No Job. Sleeping {config['ingest_wait']}s", "f": "ES Index"})
+            sleep(int(config["ingest_wait"]))
 
-    printf({"msg": "Completed Elasticsearch data load", "f": "ES Ingest"})
+    printf({"msg": "Completed Elasticsearch data load", "f": "ES Index"})
     return
 
 
 def streaming_post_to_es(client, chunk, index_name: str, type: str, job_id=None):
     success, failed = 0, 0
     try:
-        for ok, item in helpers.streaming_bulk(client, chunk, index=index_name):
+        for ok, item in helpers.parallel_bulk(client, chunk, index=index_name):
```

Comment:
How bout trying to change a couple params for better throughput? Or are you reserving that experimentation for the follow-on ticket?

```
thread_count – size of the threadpool to use for the bulk requests
chunk_size – number of docs in one chunk sent to es (default: 500)
max_chunk_bytes – the maximum size of the request in bytes (default: 100MB)
...
queue_size – size of the task queue between the main thread (producing chunks to send) and the processing threads.
```

I'm thinking at least 8 worker threads, a chunk size closer to 4000, and then putting a limiting factor of 1MB (1048576 bytes) on `max_chunk_bytes`. I haven't quite worked out how `queue_size` and `thread_count` play on each other.

Reply:
I thought about it and tested it locally a little. I couldn't find a better combination than the defaults. And that is just local to my machine, no idea what the behaviors will be like from the jenkins worker on the internal AWS network. My instict is the leave the defaults as-is for now and spend time with A-B tests in the future if there is a need. Just goin from 1 to 4 threads is going to significantly help